docs/beta/plugins/discord.md

```diff
@@ -1,5 +1,248 @@
+
+
+For instructions on setting up the bot and configuring the plugin, see the [Configuration](#configuration) section below. 
+
+
+## Cred flow
+
+Cred is minted on Discord reactions. It then flows to connected nodes.
+
+TO DO: add example with diagram....The below diagram illustrates a typical pattern of Cred flow in Discord....
+
+
+
+The full set of node and edge types used by the Discord plugin are defined below.
+
+### Nodes
+
+The Discord plugin creates the following types of nodes in the contribution graph:
+
+- **User**:
+
+A Discord user account, e.g. [@decentralion {SB: correct link?}](). User accounts do not mint Cred, so setting a node weight would have no effect. Using the identity plugin, it's possible to "collapse" user nodes with other identity nodes into a single, canonical identity. For example, if a contributor had a Discord user account and a GitHub account, then the identity plugin can collapse those identities together.
+
+Users are connected to messages they author, to messages they react to, and to messages that mention them {SB: is this true?}.
+
+- **Guild Member**:
+
+A Discord user account, e.g. [@decentralion {SB: correct?}](), that has special powers. E.g. their reactions mint more Cred {SB: totally making this up here. Is this part of beta launch, or just a hidden alpha feature for MetaGame?}
+
+- **Bot**:
+
+{SB: below is copied from GitHub page. Still true for Discord? Do we support Discord bots?}
+A Discord user account that has been explicitly marked as a bot, via inclusion in bots.js. This is useful so we can filter out bot accounts from receiving grain or showing up in the Cred rankings.
+
+Bots have the same connections as users.
+
+- **Message**:
+
+A Discord message, e.g. ["Welcome to the SourceCred community!..."](https://discordapp.com/channels/453243919774253079/715770410955964477/715776215528505385) {SB: this is a link to the welcome message in the SourceCred Discord #start-here channel}. A message node will be connected to its author (user node creating the message), emoji reactions to the message, and any references to the message {SB: is this true??}. 
+
+[INSERT SCREENSHOT MESSAGE]
+
+- **Channel**:
+
+A Discord channel, e.g. [#start-here](https://discord.gg/qadM9u) {SB: link needs checking. Got link by clicking 'Create Invite' on the channel, but getting 'invalid link' msg for some reason when I try it locally). A channel node will be connected to all messages in that channel. 
+
+Found this in the code and pasting here in case the plugin exposes these channel types. 
+
+```
+ * Discord Channels can be of various types, defined below.
+ * See: https://discordapp.com/developers/docs/resources/channel#channel-object-channel-types
+ */
+export type ChannelType =
+  | "GUILD_TEXT"
+  | "DM"
+  | "GUILD_VOICE"
+  | "GROUP_DM"
+  | "GUILD_CATEGORY"
+  | "GUILD_NEWS"
+  | "GUILD_STORE";
+```
```

Comment:
Don't think this is relevant to the actual graph, its just used internally when getting the data.

Reply:
Removed.

Reply:
Removed.